ddp-workspace/projects/ddp-basil/src/app/components/app/app.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { NoopScrollStrategy } from '@angular/cdk/overlay';
+import { MatDialog } from '@angular/material/dialog';
+import { TranslateService } from '@ngx-translate/core';
+import { CompositeDisposable, RenewSessionNotifier } from 'ddp-sdk';
+import { SessionWillExpireComponent } from 'toolkit';
+
+@Component({
+    selector: 'app-root',
+    templateUrl: './app.component.html'
+})
+export class AppComponent implements OnInit, OnDestroy {
+    private anchor = new CompositeDisposable();
+    private readonly DIALOG_BASE_SETTINGS = {
+        width: '740px',
+        position: { top: '30px' },
+        data: {},
+        autoFocus: false,
+        scrollStrategy: new NoopScrollStrategy()
+    };
+
+    constructor(
+        private dialog: MatDialog,
+        private renewNotifier: RenewSessionNotifier,
+        private translate: TranslateService) { }
+
+    ngOnInit(): void {
+        this.initSessionWillExpireListener();
+        this.initTranslate();
+    }
+
+    ngOnDestroy(): void {
+        this.anchor.removeAll();
+    }
+
+    public openSessionWillExpireDialog(): void {
```

Comment:
private? same for closeSessionWillExpireDialog

Reply:
I think these will get used in the HTML template, so I think they'll need to be public? Anyways, I didn't get time to finish implementing everything, maybe finish in next PR.